test/e2e/gardener/upgrade/upgrade.go

```diff
@@ -33,8 +33,8 @@ import (
 
 var _ = Describe("Gardener upgrade Tests for", func() {
 	var (
-		gardenerPreviousRelease = os.Getenv("GARDENER_PREVIOUS_RELEASE")
-		gardenerCurrentRelease  = os.Getenv("GARDENER_NEXT_RELEASE")
+		gardenerPreviousRelease = os.Getenv("GARDENER_PREVIOUS_VERSION")
+		gardenerCurrentRelease  = os.Getenv("GARDENER_NEXT_VERSION")
```

Comment:
```suggestion
		gardenerPreviousRelease = os.Getenv("GARDENER_PREVIOUS_RELEASE")
		gardenerPreviousVersion = os.Getenv("GARDENER_PREVIOUS_VERSION")
		gardenerNextRelease     = os.Getenv("GARDENER_NEXT_RELEASE")
		gardenerNextVersion     = os.Getenv("GARDENER_NEXT_VERSION")
```

Reply:
We can use both version and release
By this way `GARDENER_PREVIOUS_VERSION` / `GARDENER_NEXT_RELEASE` value can be some commit SHA or git references. 
``` diff
-		When("Pre-Upgrade (Gardener version:'"+gardenerPreviousRelease+"')", Ordered, Label("pre-upgrade"), func() {
+		When("Pre-Upgrade (Gardener version:'"+gardenerPreviousVersion+"', Release: '"+gardenerPreviousRelease+"')", Ordered, Label("pre-upgrade"), func() {
```

```diff
-		When("Post-Upgrade (Gardener version:'"+gardenerCurrentRelease+"')", Ordered, Label("post-upgrade"), func() {
+		When("Post-Upgrade (Gardener version:'"+gardenerNextVersion+"', Release: '"+gardenerNextRelease+"')", Ordered, Label("post-upgrade"), func() {
```

```diff 
-				Expect(f.Shoot.Status.Gardener.Version).Should(Equal(gardenerPreviousRelease))
+				Expect(f.Shoot.Status.Gardener.Version).Should(Equal(gardenerPreviousVersion))
```

Reply:
Is this really necessary? It seems a bit convoluted to have both version and release. I find the distinction between version and release difficult to understand as it is not obvious why the release could be a git commit hash while the version cannot be.

With the current adaptions you can already use a commit hash and the test will still work.

Reply:
Thanks for explaining where the values are coming from. Nevertheless, I do not see the immediate benefit in bringing this value into the test. I find it rather confusing just looking at the test to simply print out a value as part of the test description without really knowing the difference between release and version. Therefore, I would prefer to omit it and keep the test rather concise.

Reply:
Then we can update test description as **`Git version:`** WDYT ?
``` diff
-		When("Pre-Upgrade (Gardener version:'"+gardenerPreviousRelease+"')", Ordered, Label("pre-upgrade"), func() {
+		When("Pre-Upgrade (Gardener version:'"+gardenerPreviousVersion+"', Git version: '"+gardenerPreviousRelease+"')", Ordered, Label("pre-upgrade"), func() {
```